cocos/2d/CCLabel.h

```diff
@@ -197,14 +197,58 @@ class CC_DLL Label : public Node, public LabelProtocol, public BlendProtocol
     * @param text The initial text.
     * @param hAlignment Text horizontal alignment.
     * @param maxLineWidth The max line width.
-    * @param imageOffset
     *
     * @return An automatically released Label object.
     * @see setBMFontFilePath setMaxLineWidth
     */
     static Label* createWithBMFont(const std::string& bmfontPath, const std::string& text,
-        const TextHAlignment& hAlignment = TextHAlignment::LEFT, int maxLineWidth = 0,
-        const Vec2& imageOffset = Vec2::ZERO);
```

Comment:
It will break compatibility too.

Reply:
I thought I took care of all old call signatures. Please correct me if I'm wrong, but here's what I did:

The old method has this signature: 
```
Label* createWithBMFont(const std::string& bmfontPath, const std::string& text,
        const TextHAlignment& hAlignment = TextHAlignment::LEFT, int maxLineWidth = 0,
        const Vec2& imageOffset = Vec2::ZERO);
```
So, you could have called it in these ways:
```
createWithBMFont(font, text);
createWithBMFont(font, text, alignment);
createWithBMFont(font, text, alignment, maxLineWidth);
createWithBMFont(font, text, alignment, maxLineWidth, offset);

Now, with the introduction of the new API, the compiler wouldn't know which method to use because it would be ambiguous. since we're using default values in some parameters. So, to fix, here's the new API:
```
CC_DEPRECATED_ATTRIBUTE static Label* createWithBMFont(const std::string& bmfontPath, const std::string& text, const TextHAlignment& hAlignment, int maxLineWidth, const Vec2& imageOffset);

Label* createWithBMFont(const std::string& bmfontPath, const std::string& text,
        const TextHAlignment& hAlignment = TextHAlignment::LEFT, int maxLineWidth = 0);

Label* createWithBMFont(const std::string& bmfontPath, const std::string& text,
        const TextHAlignment& hAlignment, int maxLineWidth, const Rect& imageRect, bool imageRotated);

Label* createWithBMFont(const std::string& bmfontPath, const std::string& text,
        const TextHAlignment& hAlignment, int maxLineWidth, const std::string& subTextureKey);
```

So, here are the different call combinations:
```
Label* createWithBMFont(const std::string& bmfontPath, const std::string& text,
        const TextHAlignment& hAlignment = TextHAlignment::LEFT, int maxLineWidth = 0);
```
takes care of these calls:
```
createWithBMFont(font, text);
createWithBMFont(font, text, alignment);
createWithBMFont(font, text, alignment, maxLineWidth);
```
and
```
CC_DEPRECATED_ATTRIBUTE static Label* createWithBMFont(const std::string& bmfontPath, const std::string& text, const TextHAlignment& hAlignment, int maxLineWidth, const Vec2& imageOffset);
```
takes care of this:
`createWithBMFont(font, text, alignment, maxLineWidth, offset);`

The new API methods allow for these calls (no default parameters in them):
```
createWithBMFont(font, text, alignment, maxLineWidth, rect, rotation);
createWithBMFont(font, text, alignment, maxLineWidth, textureKeyString);
```

Reply:
No, it is ok. Thanks.